edx_rest_api_client/tests/test_client.py

```diff
@@ -26,6 +26,10 @@ class EdxRestApiClientTests(TestCase):
         {'url': URL, 'signing_key': SIGNING_KEY, 'username': USERNAME, 'full_name': None, 'email': EMAIL},
         {'url': URL, 'signing_key': SIGNING_KEY, 'username': USERNAME, 'full_name': FULL_NAME, 'email': None},
         {'url': URL, 'signing_key': SIGNING_KEY, 'username': USERNAME, 'full_name': None, 'email': None},
+        {'url': URL, 'signing_key': SIGNING_KEY, 'username': USERNAME},
+        {'url': URL, 'signing_key': None, 'username': USERNAME},
+        {'url': URL, 'signing_key': SIGNING_KEY, 'username': None},
+        {'url': URL, 'signing_key': None, 'username': None, 'oauth_access_token': None},
```

Comment:
TBH at this point I feel like this test should go deeper and ensure that the expected auth assignment was done (or not), instead of assuming everything is correct because `__init__` didn't fail.